api-consumption/src/main/java/org/zalando/nakadi/service/subscription/StreamingContext.java

```diff
@@ -277,10 +284,28 @@ public boolean isSubscriptionConsumptionBlocked() {
     }
 
     public boolean isConsumptionBlocked(final ConsumedEvent event) {
+        final var actualEventType = event.getPosition().getEventType();
+        if(eventTypeCache.getEventType(actualEventType).getCategory() != EventCategory.UNDEFINED) {
+            try {
+                final String metadataEventTypeName = kafkaRecordDeserializer.getEventTypeName(event.getEvent());
+                if(!actualEventType.contains(metadataEventTypeName)){
+                    LOG.warn("Found unexpected event for event type: {} " +
+                                    "but expected {} having offset {}. Event timestamp: {}, Source topic:  {}",
+                            metadataEventTypeName, event.getPosition().getEventType(), event.getPosition(),
+                            event.getTimestamp(),
+                            event.getPosition().getTimeline().getEventType());
+                    return true;
+                }
+            } catch (final IOException e) {
+                //this case shouldnt happen
+                LOG.error("Failed to parse metadata", e);
+                return false;
```

Comment:
I think it's more appropriate to let this exception propagate and be handled elsewhere.  Neither `true` nor `false` are satisfying here as return value.